src/Util/Writers/StatusWriter.php

```diff
@@ -0,0 +1,141 @@
+<?php
+/**
+ * Status Writer to send output to STDERR.
+ *
+ * ---------------------------------------------------------------------------------------------
+ * This class is intended for internal use only and is not part of the public API.
+ * This also means that it has no promise of backward compatibility. Use at your own risk.
+ * ---------------------------------------------------------------------------------------------
+ *
+ * @internal
+ *
+ * @author    Greg Sherwood <gsherwood@squiz.net>
+ * @author    Juliette Reinders Folmer <phpcs_nospam@adviesenzo.nl>
+ * @copyright 2025 PHPCSStandards and contributors
+ * @license   https://github.com/PHPCSStandards/PHP_CodeSniffer/blob/master/licence.txt BSD Licence
+ */
+
+namespace PHP_CodeSniffer\Util\Writers;
+
+final class StatusWriter
+{
+
+    /**
+     * The stream to write to.
+     *
+     * @var resource
+     */
+    private static $stream = STDERR;
+
+    /**
+     * If TRUE, requests to print a status message will be ignored.
+     *
+     * @var boolean
+     */
+    private static $paused = false;
```

Comment:
Maybe it is just me, but I'm unsure about the name of this property, `$pauseCount`, and the `pause()` method. When I see a call to `StatusWriter::pause()`, it is not immediately clear that the output will be suppressed/ignored.  The description of this property reinforces this impression. It reads "ignored" and not "paused".

`StatusWriter::pause()` doesn't just pause output. It creates a state where regular output is suppressed/ignored.

With that in mind, here are a few alternatives that occurred to me to replace the `pause()` method (and then the properties, and potentially `resume()` could be renamed following the same pattern):

- `suppressOutput()`
- `startQuietMode()`

What do you think?

Reply:
The pause/resume naming make sense to me, but I can see how one could think at first that the messages are buffered when the writer is paused and not discarded.

Maybe using `stop()` would be more explicit here ?

Reply:
I'm going to have a think about this for a moment.

Reply:
As this PR is blocking everything after, I'm going to merge it without changing for the moment.

If we come up with better names, either for the methods or the class, I'll happily have a look at a follow-up PR to this one, either before or after the 4.0 release (as this class is marked as `@internal`, so there is no BC-guarantee and we can still change it if we deem it necessary).